fw/http_msg.c

```diff
@@ -868,6 +868,19 @@ __hdr_expand(TfwHttpMsg *hm, TfwStr *orig_hdr, const TfwStr *hdr, bool append)
 	return tfw_strcpy(append ? &it : orig_hdr, hdr);
 }
 
+static void
+__hdr_del_from_tbl(TfwHttpHdrTbl *ht, unsigned int hid)
+{
+	if (hid < TFW_HTTP_HDR_RAW) {
+		TFW_STR_INIT(&ht->tbl[hid]);
```

Comment:
Seems we can't use this function for regular header removing and for trailer header removing in the same time.
For trailer we must delete only the header that located in the trailer, however this function will remove the header for both sections headers and trailer. The case is duplicated header in trailer and in headers section `TFW_STR_INIT(&ht->tbl[hid]);` will NULLify both of them, but must set to NULL only the header in trailers. We must do `TFW_STR_FOR_EACH_DUP` for such header and remove duplicate only with  `TFW_STR_TRAILER` flag.

For raw headers the same, we must check duplication before memmoving or decrementing `off`.

Reply:
@const-t @EvgeniiMekhanik it seems this worth commenting in code?